service/src/main/java/datawave/microservice/dictionary/data/DatawaveDataDictionaryImpl.java

```diff
@@ -132,14 +132,23 @@ private BatchScanner fetchResults(Connector connector, String metadataTableName,
     }
     
     private Collection<DefaultMetadataField> transformResults(Iterator<Entry<Key,Value>> iterator, Map<String,String> reverseMapping,
-                    Collection<String> dataTypeFilters) {
+                    Collection<String> dataTypeFilters, int limit, int offset) {
         HashMap<String,Map<String,DefaultMetadataField>> fieldMap = Maps.newHashMap();
         boolean shouldNotFilterDataTypes = CollectionUtils.isEmpty(dataTypeFilters);
         
         final Text holder = new Text(), row = new Text();
         
+        // Skip the number of rows specified in the query offset.
+        for (int skippedFields = 0; skippedFields < offset; skippedFields++) {
+            if (iterator.hasNext()) {
+                iterator.next();
+            }
+        }
+        
```

Comment:
I don't think this will work. If you notice on lines 107 and 110 of this file, we're passing in a batch scanner. The order of the results from a BatchScanner's iterator isn't guaranteed so while the algorithm would work for the first page of results, subsequent pages could produce duplicate results. Said another way, there's no way to get the whole dictionary in pages with this approach.

A related question... if I can only pass in offset and limit, how am I supposed to know when I've retrieved the entire dictionary?